clients/linux/src/background_work.rs

```diff
@@ -0,0 +1,120 @@
+use crate::messages::{Messenger, Msg};
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+pub struct BackgroundWork {
+    pub messenger: Messenger,
+    pub auto_save_state: AutoSaveState,
+    pub auto_sync_state: AutoSyncState,
+}
+
+impl BackgroundWork {
+    pub fn default(m: &Messenger) -> Self {
+        Self {
+            messenger: m.clone(),
+            auto_save_state: AutoSaveState::default(),
+            auto_sync_state: AutoSyncState::default(),
+        }
+    }
+
+    pub fn current_time() -> u128 {
+        SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards")
+            .as_millis()
+    }
+
+    pub fn init_background_work(state: Arc<Mutex<Self>>) {
+        loop {
+            thread::sleep(Duration::from_secs(1));
+            let mut bgs = state.lock().unwrap();
+
+            if bgs.auto_save_state.is_active {
+                bgs.auto_save();
+            }
+
+            if bgs.auto_sync_state.is_active {
+                bgs.auto_sync();
+            }
+        }
+    }
+
+    pub fn auto_sync(&mut self) {
+        let current_time = Self::current_time();
+
+        let time_since_edit = current_time - self.auto_save_state.last_change;
```

Comment:
In all the references to `time_since_...` or `time_between_...`, we should probably call all these variables `millis_since...` or `millis_between_...`